File: app/src/main/java/smt3/assignme_11/Main_Activity.java
package smt3.assignme_11;

import android.os.Bundle;

import me.ibrahimsn.lib.OnItemSelectedListener;
import me.ibrahimsn.lib.SmoothBottomBar;
import smt3.assignme_11.timeline.Fragment_2_timeline;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class Main_Activity extends AppCompatActivity {

    SmoothBottomBar smoothBottomBar;

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        smoothBottomBar = findViewById(R.id.bottomBar);

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.main_fragment, new Fragment_1_home());
        fragmentTransaction.commit();


        smoothBottomBar.setOnItemSelectedListener(new OnItemSelectedListener() {
            @Override
            public boolean onItemSelect(int i) {
                if (i==0){

                    FragmentManager fragmentManager = getSupportFragmentManager();
                    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.main_fragment, new Fragment_1_home());
                    fragmentTransaction.commit();
                }

                if (i==1){

                    FragmentManager fragmentManager = getSupportFragmentManager();
                    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.main_fragment, new Fragment_2_timeline());
                    fragmentTransaction.commit();
                }

                if (i==2){

                    FragmentManager fragmentManager = getSupportFragmentManager();
                    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.main_fragment, new Fragment_3_archive());
                    fragmentTransaction.commit();
                }

                if (i==3){

                    FragmentManager fragmentManager = getSupportFragmentManager();
                    FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                    fragmentTransaction.replace(R.id.main_fragment, new Fragment_4_setting());
                    fragmentTransaction.commit();
                }
                return false;
            }
        });
    }
}

File: app/src/main/java/smt3/assignme_11/MainActivity.kt
package smt3.assignme_11

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.NavController
import androidx.navigation.findNavController
import android.widget.Toast
import android.view.Menu
import android.view.MenuItem
import android.widget.PopupMenu
import android.view.LayoutInflater
import androidx.navigation.ui.setupActionBarWithNavController
import smt3.assignme_11.databinding.ActivityMainBinding
import androidx.navigation.fragment.findNavController


class MainActivity : AppCompatActivity() {

    private lateinit var navController: NavController
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }




}